tests/langchain/test_chat_utils.py

```diff
@@ -0,0 +1,113 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from langchain.agents import AgentExecutor
+from langchain.chat_models.base import SimpleChatModel
+from langchain.schema import AIMessage, HumanMessage, SystemMessage
+from langchain_core.messages.ai import AIMessageChunk
+
+from mlflow.langchain.chat_utils import (
+    _transform_request_json_for_chat_if_necessary,
+    _try_transform_response_iter_to_chat_format,
+    _try_transform_response_to_chat_format,
+)
+
+
+@pytest.mark.parametrize(
+    "response",
+    [
+        "string_response",
+        {"dict_response": "response"},
+        ["list_response"],
+        AIMessage(content="ai_message_response"),
+    ],
+)
+def test_transform_response_to_chat_format(response):
+    if isinstance(response, (list, dict)):
+        assert _try_transform_response_to_chat_format(response) == response
+    else:
+        if isinstance(response, str):
+            converted_response = _try_transform_response_to_chat_format(response)
+            assert isinstance(converted_response, dict)
+            assert converted_response["id"] is None
+            assert converted_response["choices"][0]["message"]["content"] == response
+        elif isinstance(response, AIMessage):
+            converted_response = _try_transform_response_to_chat_format(response)
+            assert isinstance(converted_response, dict)
+            assert converted_response["id"] == getattr(response, "id", None)
+            assert converted_response["choices"][0]["message"]["content"] == response.content
+
+
+@pytest.mark.parametrize(
+    "response",
+    [
+        ["string_response"],
+        [{"dict_response": "response"}],
+        [AIMessage(content="ai_message_response")],
+        [
+            AIMessageChunk(
+                content="ai_message_chunk_response",
+                id="123",
+                response_metadata={"finish_reason": "done"},
+            )
+        ],
+    ],
+)
+def test_transform_response_iter_to_chat_format(response):
+    converted_response = list(_try_transform_response_iter_to_chat_format(response))
+    assert len(converted_response) == 1
+    if isinstance(response[0], (list, dict)):
+        assert converted_response[0] == response[0]
+    else:
+        if isinstance(converted_response[0], str):
+            assert converted_response[0]["id"] is None
```

Comment:
I'd consider splitting this test because parameterization increases code complexity.